app/src/main/java/com/ciandt/dragonfly/example/features/download/DownloadReceiver.kt

```diff
@@ -0,0 +1,25 @@
+package com.ciandt.dragonfly.example.features.download
+
+import android.app.DownloadManager
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+
+class DownloadReceiver : BroadcastReceiver() {
+
+    override fun onReceive(context: Context, intent: Intent) {
+
+        val id = intent.getLongExtra(DownloadManager.EXTRA_DOWNLOAD_ID, 0)
+        if (id !in DownloadHelper.getDownloads(context)) {
```

Comment:
Changed to isValid.

getDownloadedFile() dont return null. If ID is not in download manager database, it is because download was cancelled.